src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -27251,6 +27264,128 @@ static void M1<T>(T t1)
             comp2.VerifyDiagnostics(expected);
         }
 
+        [Fact]
+        public void AllowNull_Property_WithNotNull()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class COpen<TOpen>
+{
+    [AllowNull, NotNull]public TOpen P1 { get; set; } = default!;
+}
+public class CClass<TClass> where TClass : class
+{
+    [AllowNull, NotNull]public TClass P2 { get; set; } = null!;
```

Comment:
why use `null!` here?

Reply:
I think a comment later on in the tests clued me in :)

---
In reply to: [323000597](https://github.com/dotnet/roslyn/pull/38610#discussion_r323000597) [](ancestors = 323000597)

Reply:
Also noticed that we have `error CS8050: Only auto-implemented properties can have initializers.`. So we can be certain that an initializer is not causing a 'set' call which does some validation/transformation. It will write the initializer value straight to the backing field.

---
In reply to: [323001615](https://github.com/dotnet/roslyn/pull/38610#discussion_r323001615) [](ancestors = 323001615,323000597)